Source/JavaScriptCore/yarr/YarrErrorCode.h

```diff
@@ -64,6 +65,7 @@ enum class ErrorCode : uint8_t {
     InvalidUnicodePropertyExpression,
     TooManyDisjunctions,
     OffsetTooLarge,
+    OutOfMemory,
```

Comment:
I think it makes sense to add this error to hasHardError() below as another error that is not a hard (syntax) error.

Reply:
Fixed.  I've updated the enum list to be clearer.  But in the end, `OutOfMemory` was removed since it's no longer needed.